### PR TITLE
Refactor ensure resource

### PR DESCRIPTION
**Pull request by mdellweg**

### CODE REVIEW

File: plugins/module_utils/foreman_helper.py [hunk 1]
```diff
@@ -286,8 +303,10 @@ def __init__(self, argument_spec, **kwargs):
             state=dict(choices=['present', 'absent'], default='present'),
         )
         args.update(argument_spec)
+        name_map = kwargs.pop('name_map', {})
```
Comment (evgeni) [inline plugins/module_utils/foreman_helper.py]: Do we really want to touch the `nailgun` API?

File: plugins/modules/katello_sync_plan.py [hunk 2]
```diff
@@ -143,7 +143,7 @@ def main():
 
     sync_plan = find_sync_plan(module, entity_dict['name'], entity_dict['organization'], failsafe=True)
 
-    entity_dict = sanitize_entity_dict(entity_dict, name_map)
+    entity_dict = sanitize_entity_dict(entity_dict, name_map=name_map)
```
Comment (evgeni) [inline plugins/modules/katello_sync_plan.py]: why this change? (besides that I am not sure I want to touch the nailgun modules at all), `sanitize_entity_dict` takes exactly two params :)

**Comment by evgeni** [comment]
Overall, I like :)

File: plugins/modules/katello_sync_plan.py (hunk 2 above)
Comment (mdellweg) [reply plugins/modules/katello_sync_plan.py]: `sed -i '...' *.py`
I can take it back. ;)

File: plugins/module_utils/foreman_helper.py (hunk 1 above)
Comment (mdellweg) [reply plugins/module_utils/foreman_helper.py]: You did for the documentation. But, you are right. There is no benefit.

I thought about moving the not nailgun Classes to the cement file, because i accidentally added my changes there, and wondered, nothing happend.

Comment (evgeni) [reply plugins/module_utils/foreman_helper.py]: I did, because it was a no-code change ;)

File: plugins/modules/katello_sync_plan.py (hunk 2 above)
Comment (evgeni) [reply plugins/modules/katello_sync_plan.py]: I'd prefer not to touch these files.
